parts/linux/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -253,14 +253,18 @@ retrycmd_get_tarball_from_registry_with_oras() {
 retrycmd_get_access_token_for_oras() {
     retries=$1; wait_sleep=$2; url=$3
     for i in $(seq 1 $retries); do
-        ACCESS_TOKEN_OUTPUT=$(timeout 60 curl -v -s -H "Metadata:true" --noproxy "*" "$url")
-        if [ -n "$ACCESS_TOKEN_OUTPUT" ]; then 
+        response=$(timeout 60 curl -v -s -H "Metadata:true" --noproxy "*" "$url" -w "\n%{http_code}")
+        ACCESS_TOKEN_OUTPUT=$(echo "$response" | sed '$d')
+        http_code=$(echo "$response" | tail -n1)
+        if [ -n "$ACCESS_TOKEN_OUTPUT" ] && [ "$http_code" -eq 200 ]; then 
             echo "$ACCESS_TOKEN_OUTPUT"
             return 0
         fi
         sleep $wait_sleep
     done
-    return $ERR_ORAS_PULL_NETWORK_TIMEOUT
+    echo "connect to IMDS to get access token timeout"
+    http_code=${http_code:-$ERR_ORAS_IMDS_TIMEOUT}
```

Comment:
you are. mixing HTTP error code with CSE error code here, this could also be confusing,  seeing 210 I would expect this is a valid HTTP_CODE.  I would log something more obvious.  If http_code is set, we log a message saying fail to reach IMDS after after X retires due to error code XYZand we return a CSE ERROR for CURL FAILURE,  if there is no error code, we log a TIMEOUT ERROR and we return a CSE ERROR for TIMEOUT.